routers/repo/issue.go

```diff
@@ -1821,80 +1811,97 @@ func updatePullReviewRequest(ctx *context.Context) {
 	}
 
 	for _, issue := range issues {
-		if issue.IsPull {
-
-			if reviewID < 0 {
-				if err := issue.LoadRepo(); err != nil {
-					ctx.ServerError("issue.LoadRepo", err)
-					return
-				}
-
-				if err := issue.Repo.GetOwner(); err != nil {
-					ctx.ServerError("issue.Repo.GetOwner", err)
-					return
-				}
-
-				if issue.Repo.Owner.IsOrganization() {
-					team, err := models.GetTeamByID(-reviewID)
-					if err != nil {
-						ctx.ServerError("models.GetTeamByID", err)
-						return
-					}
-
-					if team.OrgID != issue.Repo.OwnerID {
-						ctx.Status(403)
-						return
-					}
-
-					err = isValidTeamReviewRequest(team, ctx.User, action == "attach", issue)
-					if err != nil {
-						ctx.Status(403)
-						return
-					}
+		if err := issue.LoadRepo(); err != nil {
+			ctx.ServerError("issue.LoadRepo", err)
+			return
+		}
 
-					err = issue_service.TeamReviewRequest(issue, ctx.User, team, action == "attach")
-					if err != nil {
-						ctx.ServerError("ReviewRequest", err)
-						return
-					}
+		if !issue.IsPull {
```

Comment:
I think it's better to do this check before ``issue.LoadRepo``, because if this issue is not a pull, it's not necessary to load repo.

Reply:
Ah but the logging message requires the repo.

In this case because we never expect to see an attempt to review an issue we actually want to have a decent log message because something is wrong - either it's a misbehaving user or more likely a bug. Almost all of the time we're going to have to load the repo - this should be an extremely rare situation - and in fact I wouldn't be surprised if it wasn't already loaded.

Sometimes it's worth doing the extra work just to ensure that we get decent messages.